packages/openapi3/test/http-specs/specs.test.ts

```diff
@@ -0,0 +1,166 @@
+import { resolvePath } from "@typespec/compiler";
+import { findTestPackageRoot } from "@typespec/compiler/testing";
+import { deepStrictEqual, ok, strictEqual } from "assert";
+import { exec, execSync } from "child_process";
+import { afterAll, beforeAll, describe, expect, it } from "vitest";
+import { worksFor } from "./../works-for.js";
+import {
+  checkServe,
+  defineSpecTests,
+  markCoverage,
+  validataDataWithSchema,
+} from "./utils/spec-snapshot-testing.js";
+
+const pkgRoot = await findTestPackageRoot(import.meta.url);
+const specsRoot = resolvePath(pkgRoot, "node_modules", "@typespec", "http-specs", "specs");
+const rootOutputDir = resolvePath(pkgRoot, "test", "http-specs", "output");
+
+describe.skip("http-specs convert", () => {
+  defineSpecTests({
+    specDir: specsRoot,
+    outputDir: rootOutputDir,
+    // Excluded cases hits errors/warnings during conversion.
+    exclude: [
+      // warning @typespec/openapi3/invalid-format: Collection format 'tsv' is not supported in OpenAPI3 query parameters. Defaulting to type 'string'.
+      // > 49 |     colors: string[],
+      "parameters/collection-format",
+
+      // warning @typespec/openapi3/xml-unwrapped-invalid-property-type: XML `@unwrapped` can only used on array properties or primitive ones in the OpenAPI 3 emitter, Property 'content' will be ignored.
+      // > 70 |   @unwrapped content: string;
+      "payload/xml",
+
+      // error @typespec/openapi3/unsupported-status-code-range: Status code range '494 to '499' is not supported. OpenAPI 3.0 can only represent range 1XX, 2XX, 3XX, 4XX and 5XX. Example: `@minValue(400) @maxValue(499)` for 4XX.
+      // 59 | model ErrorInRange {
+      "response/status-code-range",
+
+      // error @typespec/openapi3/path-query: OpenAPI does not allow paths containing a query string.
+      // > 435 |       op array(param: string[]): void;
+      "routes",
+    ],
+  });
+});
+
+describe("http-specs cases", () => {
+  beforeAll(async () => {
+    exec("pnpm spector-serve", { cwd: pkgRoot });
+    await checkServe();
+  });
+
+  afterAll(() => {
+    execSync("pnpm spector-stop", { stdio: "inherit", cwd: pkgRoot });
+  });
+  worksFor(["3.0.0", "3.1.0"], ({ openApiForFile }) => {
+    describe("AuthApiKeyClient Rest Client", () => {
```

Comment:
I fail to see the real value in this PR. Do we really need this? You go and reimplement everything to validate pretty much nothing. This is a lot of wasted effort.

What is the motivation behind this?

Reply:
The goal is to leverage the full test specs in Spector to verify various scenarios for the OAS3 emitter and display the results on the Spector dashboard for scenario coverage. This way, any changes in TypeSpec/Emitter can be quickly identified for potential breakings.  

But I am not quite get the point in this PR about how we validate the result.